micro/walk.go

```diff
@@ -36,16 +36,15 @@ for example:
 
 	assv v s <==> (assp (λ (v) (var=? u v)) s))
 */
-func assv(v *ast.Variable, ss Substitutions) (*Substitution, bool) {
+func assv(v *ast.Variable, ss Substitutions) (*ast.SExpr, bool) {
 	if ss == nil {
 		return nil, false
 	}
-	for i, s := range ss {
-		if v.Name == s.Var {
-			return ss[i], true
-		}
-	}
-	return nil, false
+    value, ok := ss[v.Name]
```

Comment:
Aha, so here is where we get the speed increase, right?

Reply:
With one caveat: assoc list is faster for very recently introduced vars.
In practise should mean smth like: very small slowdown for super simple cases,
significant speedup for large cases.